src/templates/layouts/healthCareLocalFacility/HealthServices/ServiceAddress.test.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import { ServiceAddress } from './ServiceAddress'
+import { FieldAddress } from '@/types/drupal/field_type'
+import { ParagraphServiceLocationAddress } from '@/types/drupal/paragraph'
+
+describe('ServiceAddress', () => {
+  const mockFacilityAddress: FieldAddress = {
+    langcode: 'en',
+    country_code: 'US',
+    administrative_area: 'CA',
+    locality: 'Los Angeles',
+    postal_code: '90001',
+    address_line1: '123 Main St',
+  }
+
+  const mockServiceAddress: FieldAddress = {
+    langcode: 'en',
+    country_code: 'US',
+    administrative_area: 'CA',
+    locality: 'San Francisco',
+    postal_code: '94101',
+    address_line1: '456 Service Rd',
+  }
+
+  const mockServiceLocationAddress: ParagraphServiceLocationAddress = {
+    drupal_internal__id: 1,
+    drupal_internal__revision_id: 1,
+    langcode: 'en',
+    status: true,
+    id: '1',
+    type: 'paragraph--service_location_address',
+    field_address: mockServiceAddress,
+    field_building_name_number: 'Building 5',
+    field_clinic_name: 'Cardiology Clinic',
+    field_use_facility_address: false,
+    field_wing_floor_or_room_number: 'Room 203',
+  }
+
+  it('returns null when no address data exists', () => {
+    const { container } = render(
+      <ServiceAddress
+        serviceLocationAddress={{
+          ...mockServiceLocationAddress,
+          field_address: null,
+          field_clinic_name: '',
+          field_building_name_number: '',
+          field_wing_floor_or_room_number: '',
+        }}
+      />
+    )
+    expect(container.firstChild).toBeNull()
+  })
+
+  it('displays no address when useFacilityAddress is true', () => {
+    render(
+      <ServiceAddress
+        serviceLocationAddress={{
+          ...mockServiceLocationAddress,
+          field_use_facility_address: true,
+        }}
+        facilityAddress={mockFacilityAddress}
+      />
+    )
+
+    expect(
+      screen.getByRole('heading', { name: 'Cardiology Clinic', level: 4 })
+    ).toBeInTheDocument()
+    // Neither the facility nor the service address should be displayed
+    expect(
+      screen.queryByText(mockFacilityAddress.address_line1)
+    ).not.toBeInTheDocument()
+    expect(
+      screen.queryByText(mockServiceAddress.address_line1)
+    ).not.toBeInTheDocument()
+  })
+
+  it('displays service address when useFacilityAddress is false', () => {
+    render(
+      <ServiceAddress
+        serviceLocationAddress={{
+          ...mockServiceLocationAddress,
+          field_use_facility_address: false,
+        }}
+      />
+    )
+
+    expect(screen.getByText('Cardiology Clinic')).toBeInTheDocument()
+    expect(screen.getByText('456 Service Rd')).toBeInTheDocument()
+    expect(screen.getByText('San Francisco, CA 94101')).toBeInTheDocument()
+    expect(screen.getByText('Building 5')).toBeInTheDocument()
+    expect(screen.getByText('Room 203')).toBeInTheDocument()
```

Comment:
AI filled this in. I was going to replace it with the variables we defined above, but...it was really wordy and annoying. 🙈 (See the test above.)

If you have strong opinions, I can replace these with `mockFacilityAddress.address_line1` and the like.

Reply:
this is also fine with me. More of a chore if we need to update things later, but still an accurate test as it is.